packages/nuxt/src/core/runtime/nitro/handlers/renderer.ts

```diff
@@ -134,17 +134,24 @@ export default defineRenderHandler(async (event): Promise<Partial<RenderResponse
     ? getQuery(event) as unknown as NuxtPayload['error'] & { url: string }
     : null
 
-  if (ssrError && ssrError.statusCode) {
-    ssrError.statusCode = Number.parseInt(ssrError.statusCode as any)
-  }
-
   if (ssrError && !('__unenv__' in event.node.req) /* allow internal fetch */) {
     throw createError({
       statusCode: 404,
       statusMessage: 'Page Not Found: /__nuxt_error',
     })
   }
 
+  if (ssrError) {
+    ssrError.statusCode &&= Number.parseInt(ssrError.statusCode as any)
+    if (typeof ssrError.data === 'string' && ssrError.data.startsWith('{')) {
```

Comment:
WDYT about extending this condition to include arrays (valid JSON), or remove it (if not necessary)?
```suggestion
    if (typeof ssrError.data === 'string') {
```

Reply:
Well, coderabbitai was first 😄